examples/animation/custom_skinned_mesh.rs

```diff
@@ -0,0 +1,138 @@
+            brightness: 1.0,
+            ..Default::default()
+        })
+        .add_startup_system(setup.system())
+        .add_system(joint_animation.system())
+        .run();
+}
+
+struct AnimatedJoint;
+
+fn setup(
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+    mut skinned_mesh_inverse_bindposes_assets: ResMut<Assets<SkinnedMeshInverseBindposes>>,
+) {
+    // Create a camera
+    let mut camera = OrthographicCameraBundle::new_2d();
+    camera.orthographic_projection.near = -1.0;
+    camera.orthographic_projection.far = 1.0;
+    camera.orthographic_projection.scale = 0.005;
+    camera.transform = Transform::from_xyz(0.0, 1.0, 0.0);
+    commands.spawn_bundle(camera);
+
+    // Create inverse bindpose matrices for a skeleton consists of 2 joints
+    let inverse_bindposes =
+        skinned_mesh_inverse_bindposes_assets.add(SkinnedMeshInverseBindposes(vec![
+            Mat4::from_translation(Vec3::new(-0.5, -1.0, 0.0)),
+            Mat4::from_translation(Vec3::new(-0.5, -1.0, 0.0)),
+        ]));
+
+    // Create a mesh
+    let mut mesh = Mesh::new(PrimitiveTopology::TriangleList);
+    mesh.set_attribute(
+        Mesh::ATTRIBUTE_POSITION,
+        vec![
+            [0.0, 0.0, 0.0],
+            [1.0, 0.0, 0.0],
+            [0.0, 0.5, 0.0],
+            [1.0, 0.5, 0.0],
+            [0.0, 1.0, 0.0],
+            [1.0, 1.0, 0.0],
+            [0.0, 1.5, 0.0],
+            [1.0, 1.5, 0.0],
+            [0.0, 2.0, 0.0],
+            [1.0, 2.0, 0.0],
+        ],
+    );
+    mesh.set_attribute(Mesh::ATTRIBUTE_NORMAL, vec![[0.0, 0.0, 1.0]; 10]);
+    mesh.set_attribute(Mesh::ATTRIBUTE_UV_0, vec![[0.0, 0.0]; 10]);
+    mesh.set_attribute(
+        "Vertex_JointIndex",
+        vec![
+            [0, 0, 0, 0],
+            [0, 0, 0, 0],
+            [0, 1, 0, 0],
+            [0, 1, 0, 0],
+            [0, 1, 0, 0],
+            [0, 1, 0, 0],
+            [0, 1, 0, 0],
+            [0, 1, 0, 0],
+            [0, 1, 0, 0],
+            [0, 1, 0, 0],
+        ],
+    );
+    mesh.set_attribute(
+        "Vertex_JointWeight",
+        vec![
+            [1.00, 0.00, 0.0, 0.0],
+            [1.00, 0.00, 0.0, 0.0],
+            [0.75, 0.25, 0.0, 0.0],
+            [0.75, 0.25, 0.0, 0.0],
+            [0.50, 0.50, 0.0, 0.0],
+            [0.50, 0.50, 0.0, 0.0],
+            [0.25, 0.75, 0.0, 0.0],
+            [0.25, 0.75, 0.0, 0.0],
+            [0.00, 1.00, 0.0, 0.0],
+            [0.00, 1.00, 0.0, 0.0],
+        ],
+    );
+    mesh.set_indices(Some(Indices::U16(vec![
+        0, 1, 3, 0, 3, 2, 2, 3, 5, 2, 5, 4, 4, 5, 7, 4, 7, 6, 6, 7, 9, 6, 9, 8,
+    ])));
+
+    // Create joint entities
+    let joint_0 = commands
+        .spawn_bundle((
+            Transform::from_xyz(0.0, 1.0, 0.0),
+            GlobalTransform::identity(),
+        ))
+        .id();
+    let joint_1 = commands
+        .spawn_bundle((
+            AnimatedJoint,
```

Comment:
TBH I think you should make and then use a `AnimatedJointBundle` rather than manually constructing it.

Reply:
I think the example is simple enough to not make a separate bundle.

Reply:
End users will want to do the same thing though right? Especially with nested bundles it seems useful.

Reply:
From my perspective, within the scope of this example, writing code like this helps users differentiate `joint_1` from `joint_0`. Both joints have `Transform` and `GlobalTransform`, while only one has `AnimatedJoint` and the other doesn't. If we have define `AnimatedJointBundle`, then do we have to create `StaticJointBundle`/`UnanimatedJointBundle` to make it coherent as well?

Reply:
Moreover, `AnimatedJoint` is just a marker struct to help the animation system filtering out the entities. Bundle is not so useful for simple use case if we have to define a bundle for each marker struct. That's why I think it is unecessary in this example.

On the other hand, I can think of a use case where a bundle is useful. It is to use enum as marker component.

```rust
enum Animation {
    Idle,
    Walk,
    Run,
}

#[derive(Bundle)]
struct AnimationBundle {
    animation: Animation,
    transform: Transform,
    global_transform: GlobalTransform,
}
```

We can have many animation as we want and we only need a single bundle for them. Then we can have a system that decides what to do based on different values of `Animation`. This is an example where having a bundle is useful.